File: lab2no3.swift
import Foundation

func checkTimeFormat(num: Int) -> Bool {
    if num < 100 {
        return false
    }
    
    let hours = num / 100
    let minutes = num % 100
    
    return (0...23).contains(hours) && (0...59).contains(minutes)
}

print("Enter the number of test cases:")
if let caseCount = readLine(), let count = Int(caseCount) {
    var i = 0
    while i < count {
        print("Enter time format (hhmm) or -1 to exit:")
        if let input = readLine(), let num = Int(input) {
            if num == -1 {
                break
            }
            
            if checkTimeFormat(num: num) {
                print("YES")
            } else {
                print("NO")
            }
            
            i += 1
        }
    }
}

File: lab2no3.py
def check_time_format(num):
    if num < 100:
        return False
    hours = num // 100
    minutes = num % 100

    return 0 <= hours <= 23 and 0 <= minutes <= 59

def main():
    case_count = int(input("Enter the number of test cases: "))
    i = 0

    while i < case_count:
        num = int(input("Enter time format (hhmm) or -1 to exit: "))
        if num == -1:
            break
        if check_time_format(num):
            print("YES")
        else:
            print("NO")
        i += 1

main()
